src/parser/mod.rs

```diff
@@ -3615,13 +3615,15 @@ impl<'a> Parser<'a> {
         if has_table {
             let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
             let table_name = self.parse_object_name(false)?;
-            if self.peek_token().token == Token::EOF {
-                Ok(Statement::UNCache {
+            match self.peek_token() {
+                TokenWithLocation {
+                    token: Token::EOF | Token::SemiColon,
+                    ..
+                } => Ok(Statement::UNCache {
                     table_name,
                     if_exists,
-                })
-            } else {
-                self.expected("an `EOF`", self.peek_token())
+                }),
+                token_loc => self.expected("`EOF` or `;`", token_loc),
             }
         } else {
             self.expected("a `TABLE` keyword", self.peek_token())
```

Comment:
(not necessarily related to this PR) looks like we can avoid the has_table condition altogether by using `self.expect_keyword` instead of `self.parse_keyword` to parse the `TABLE` keyword? if so could we make that change to simplify the function in the same go (assuming its a straightforward change)

Reply:
Oh nice catch! I can fix this in this PR if this is straightforward, otherwise I can open up a new issue.